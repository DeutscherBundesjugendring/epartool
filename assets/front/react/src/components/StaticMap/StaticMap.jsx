import React from 'react';

/* global osmStaticMapUrlTemplate */

const StaticMap = props => (
  <div className="map-static map-static-landscape">
    <img
      width={props.width}
      height={props.height}
      src={
        osmStaticMapUrlTemplate.replace(/__latitude__/g, props.latitude)
          .replace(/__longitude__/g, props.longitude)
          .replace('__width__', props.width)
          .replace('__height__', props.height)
      }
      alt={`GPS: ${props.latitude}, ${props.longitude}`}
    />
  </div>
);

StaticMap.defaultProps = {
  width: 240,
  height: 120,
};

StaticMap.propTypes = {
  latitude: React.PropTypes.number.isRequired,
  longitude: React.PropTypes.number.isRequired,
  width: React.PropTypes.number,
  height: React.PropTypes.number,
};

export default StaticMap;
