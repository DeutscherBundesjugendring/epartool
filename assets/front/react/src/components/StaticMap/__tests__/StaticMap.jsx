import { shallow } from 'enzyme';
import React from 'react';
import StaticMap from '../StaticMap';

global.osmStaticMapUrlTemplate = '%s?center=__latitude__,__longitude__&zoom=8&size=__width__x__height__&markers=__latitude__,__longitude__';

describe('rendering', () => {
  it('renders correctly with default size', () => {
    const tree = shallow(
      <StaticMap
        latitude={52.50828}
        longitude={13.38581}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with specified size', () => {
    const tree = shallow(
      <StaticMap
        latitude={52.50828}
        longitude={13.38581}
        width={150}
        height={100}
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
