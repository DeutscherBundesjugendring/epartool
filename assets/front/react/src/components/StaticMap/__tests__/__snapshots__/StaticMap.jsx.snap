// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with default size 1`] = `
<div
  className="map-static map-static-landscape"
>
  <img
    alt="GPS: 52.50828, 13.38581"
    height={120}
    src="%s?center=52.50828,13.38581&zoom=8&size=240x120&markers=52.50828,13.38581"
    width={240}
  />
</div>
`;

exports[`rendering renders correctly with specified size 1`] = `
<div
  className="map-static map-static-landscape"
>
  <img
    alt="GPS: 52.50828, 13.38581"
    height={100}
    src="%s?center=52.50828,13.38581&zoom=8&size=150x100&markers=52.50828,13.38581"
    width={150}
  />
</div>
`;
